### PR TITLE
make simpleFramingProtocol decoder and encoder public

### PR DESCRIPTION
**Pull request by wlk**
addresses #19500

Extracted encoder and decoder outside of `simpleFramingProtocol`

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala [hunk 1]
```diff
@@ -74,22 +74,29 @@ object Framing {
...
-    })
-
-    BidiFlow.fromFlowsMat(encoder, decoder)(Keep.left)
+    BidiFlow.fromFlowsMat(simpleFramingProtocolEncoder(maximumMessageLength), simpleFramingProtocolDecoder(maximumMessageLength))(Keep.left)
   }
 
+  /**
+    * Protocol decoder that is used by [[Framing#simpleFramingProtocol]]
+    */
+  def simpleFramingProtocolDecoder(maximumMessageLength: Int) = lengthField(4, 0, maximumMessageLength + 4, ByteOrder.BIG_ENDIAN).map(_.drop(4))
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala]: explicit return type, please

**Comment by patriknw** [comment]
LGTM after the return type

**Comment by wlk** [comment]
explicit return type added

**Comment by ktoso** [comment]
LGTM, please squash

**Comment by ktoso** [comment]
OK TO TEST

**Comment by wlk** [comment]
squashed and pushed

can you clarify commit message format I have seen everywhere else:
* IDK what `=` stands for (there are also other symbols)
* was able to guess that `str` is for the `akka-streams` submodule - is this correct?

**Comment by ktoso** [comment]
We're not as strict for using the `=str` prefix nowadays, I personally like it, story is: Historically we've required it to be always present, but it confused contributors so we don't require it nowadays.

The semantics are:

```
prefix = mod ~ pro
mod = "+" | "-" | "=" | "!" # new api, remove api, no api change, breaking api change
pro = prefix(module name) # so str, clu or act would be such examples
```

It's mostly to know what modules a commit touched. But as I said, we're not very strict about them nowadays.

**Comment by ktoso** [comment]
LGTM, thanks!

**Pull request merged**